### PR TITLE
Add with-unknown-type

### PR DESCRIPTION
**Pull request by david-christiansen**
This is half of #60. See that issue for a description of the new operator.

### CODE REVIEW

File: src/Expander/Monad.hs [hunk 1]
```diff
@@ -213,12 +213,14 @@ newtype ExpansionEnv = ExpansionEnv (Map.Map Binding EValue)
...
 data EValue
-  = EPrimExprMacro (Ty -> SplitCorePtr -> Syntax -> Expand ()) -- ^ For special forms
+  = EPrimExprMacro (Ty -> SplitCorePtr -> Syntax -> Expand ())
+    -- ^ For special forms
   | EPrimTypeMacro (SplitTypePtr -> Syntax -> Expand ()) (TypePatternPtr -> Syntax -> Expand ())
     -- ^ For type-level special forms - first as types, then as type patterns
   | EPrimModuleMacro (Syntax -> Expand ())
   | EPrimDeclMacro (DeclTreePtr -> DeclOutputScopesPtr -> Syntax -> Expand ())
   | EPrimPatternMacro (Either (Ty, Ty, PatternPtr) (Ty, TypePatternPtr) -> Syntax -> Expand ())
+  | EPrimUniversalMacro (MacroDest -> Syntax -> Expand ())
```
Comment (langston-barrett) [inline src/Expander/Monad.hs:223]: What is a "universal" macro?

**Comment by david-christiansen** [comment]
Universal in the sense that it works in any syntactic context, be it
expression, type, decl, etc.

File: src/Expander/Monad.hs (hunk 1 above)
Comment (david-christiansen) [reply src/Expander/Monad.hs:223]: Universal in the sense that it works in any syntactic context, be it expression, type, decl, etc.

Comment (langston-barrett) [reply src/Expander/Monad.hs:223]: What is its meaning in a pattern context? Looks like an error? And in a declaration context? What about a module context? (Not trying to be annoying, just trying to understand :smile:)

**Pull request merged**